### PR TITLE
Add missing features

### LINKED ISSUE #1: Allow specifying scope
Need a way to specify scope. `*:*:*` won't be allowed at some point

### PR DESCRIPTION
**Pull request by d2l-joelp**
US70984 - Improve d2l-ajax component
https://rally1.rallydev.com/#/51191528503d/detail/userstory/55238207600

Resolves #1, #2, and #3

### CODE REVIEW

**Comment by d2l-joelp** [comment]
@ryantmer

**Comment by omsmith** [comment]
https://github.com/Brightspace/frau-jwt/commit/57009f89f0ed69be175720799ba6da98421da989

**Comment by omsmith** [comment]
Is there no way to use _frau-jwt_ directly? If it was `XMLHttpRequest` and not superagent so it was lighter, for instance. Would there be a point in avoiding it? Similarly for _frau-xsrf-token_. Are these `iron-` things providing benefits?

**Comment by d2l-joelp** [comment]
You can use Browserify to create a standalone script with a global export, but that creates a fairly large file (I believe it was around 50KB) due to all the dependencies. But yeah, if it wasn't using superagent then I imagine the file size would be much smaller. 

The `iron-` things don't provide any benefit feature wise other than they're implemented as web components and if any other web components use the same dependencies we can properly de-dupe them and build a common shared.html file like we are doing in https://github.com/Brightspace/brightspace-integration.

File: d2l-ajax.html [hunk 1]
```diff
@@ -78,18 +87,49 @@
                     notify: true,
                     readOnly: true
                 },
+                scope: {
+                    type: String,
+                    value: '*:*:*'
```
Comment (ryantmer) [inline d2l-ajax.html]: Can the default be something other than all-access? Not sure if there are any security considerations about this or anything, granted

File: package.json [hunk 2]
```diff
@@ -1,18 +1,23 @@
 {
-  "name": "d2l-ajax-ui",
+  "name": "d2l-ajax",
   "version": "0.1.0",
```
Comment (ryantmer) [inline package.json]: Minor (or maybe v1?) bump after this is merged?

**Comment by ryantmer** [comment]
LGTM :+1: I haven't checked it out and tested it granted, should I do so?

File: package.json (hunk 2 above)
Comment (d2l-joelp) [inline package.json]: Agreed. Will bump to v1.

File: d2l-ajax.html (hunk 1 above)
Comment (d2l-joelp) [inline d2l-ajax.html]: As per our discussion I'm going to leave this default for now to be consistent with the original library. If all-access becomes not allowed in the future we can change this.

**Comment by d2l-joelp** [comment]
@ryantmer Sure, if you have time, please do.

**Comment by ryantmer** [comment]
Okeydoke, tested it up, all seems good. My only thought is that the demo should perhaps be updated to include some of this new functionality, but this I'm like 10% on that since a demo for something this "behind-the-scenes"-y is not overly useful anyway. Merge away!

**Pull request merged**